src/openqaoa-pyquil/backends/qaoa_pyquil_qpu.py

```diff
@@ -229,21 +243,38 @@ def parametric_qaoa_circuit(self) -> Program:
             # using the list above, construct the circuit
             for each_tuple in decomposition:
                 gate = each_tuple[0]()
-                qubits, rotation_angle = each_tuple[1]
+                if len(each_tuple[1]) == 1:
+                    qubits, rotation_angle = each_tuple[1][0], None
+                elif len(each_tuple[1]) == 2:
+                    qubits, rotation_angle = each_tuple[1]
+                else:
+                    raise ValueError(f"Specified an incorrect gate decomposition {each_tuple[1]}")
                 if isinstance(qubits, list):
                     new_qubits = [self.qubit_mapping[qubit] for qubit in qubits]
                 else:
                     new_qubits = self.qubit_mapping[qubits]
-                parametric_circuit = gate.apply_pyquil_gate(
-                    new_qubits, rotation_angle, parametric_circuit
-                )
+                if rotation_angle is None:
+                    parametric_circuit = gate.apply_pyquil_gate(
+                        new_qubits, parametric_circuit
+                    ) 
+                else:
+                    parametric_circuit = gate.apply_pyquil_gate(
+                        new_qubits, rotation_angle, parametric_circuit
+                    )
 
         if self.append_state:
             parametric_circuit += self.append_state
-
-        # Measurement instructions
-        for i, qubit in enumerate(self.qureg):
-            parametric_circuit += gates.MEASURE(self.qubit_mapping[qubit], ro[i])
+            
+        if self.final_mapping is None:
+            for i,qbit in enumerate(self.problem_reg):
+                parametric_circuit += gates.MEASURE(qbit, ro[i])
+        else:
+            # Measurement instructions
+            for qubit in self.qureg:
+                reorderd_qubit = self.final_mapping[qubit]
+                if reorderd_qubit in self.problem_reg:
```

Comment:
If the `reorderd_qubit` is not in the `problem_reg`, the measurement instruction is skipped. Maybe we should add some kind of error/exception here since its an unexpected behaviour

Reply:
That is exactly the behaviour we want. We need to skip the measurement operation on the qubits that are not `problem_qubits`.

Reply:
Ok Sounds good.